awx_collection/tests/integration/targets/tower_export/tasks/main.yml

```diff
@@ -0,0 +1,77 @@
+---
+- name: Generate a random string for test
+  set_fact:
+    test_id: "{{ lookup('password', '/dev/null chars=ascii_letters length=16') }}"
+  when: test_id is not defined
+
+- name: Generate names
+  set_fact:
+    org_name1: "AWX-Collection-tests-tower_export-organization-{{ test_id }}"
+    org_name2: "AWX-Collection-tests-tower_export-organization2-{{ test_id }}"
+    inventory_name1: "AWX-Collection-tests-tower_export-inv1-{{ test_id }}"
+
+- block:
+    - name: Create some organizations
+      tower_organization:
+        name: "{{ item }}"
+      loop:
+        - "{{ org_name1 }}"
+        - "{{ org_name2 }}"
+
+    - name: Create an inventory
+      tower_inventory:
+        name: "{{ inventory_name1 }}"
+        organization: "{{ org_name1 }}"
+
+    - name: Export all tower assets
+      tower_export:
+        all: true
+      register: all_assets
```

Comment:
The kind of failure I'm seeing for this is:

```
  TASK [tower_export : Export all tower assets] **********************************
  ok: [testhost]
  
  TASK [tower_export : assert] ***************************************************
  fatal: [testhost]: FAILED! => {"msg": "The conditional check 'all_assets['assets']['organizations'] | length() >= 2' failed. The error was: error while evaluating conditional (all_assets['assets']['organizations'] | length() >= 2): 'dict object' has no attribute 'assets'"}
```

It is possible that this failure is due to missing the `awxkit` library.

Is there any logic for detection of importability of `awxkit`? I would like to have a minimal check for that and give an error message that makes sense.

Reply:
unrelated but im a bit worried about exporting everything in our tests, this could add a significant amount of time to our testing, we might want to have a discussion about how we can export less while feeling we have enough coverage

Reply:
@AlanCoding There is logic in both tower_awxkit.py:
```
        if not HAS_AWX_KIT:
            self.exit_json(msg=missing_required_lib('awxkit'))
```

As well as more specific logic in tower_import/export:
```
    if not HAS_EXPORTABLE_RESOURCES:
        module.fail_json(msg="Your version of awxkit does not have import/export")
```

I'm not sure what would cause an `ok` with no `assets` entry, can you have it dump out the all_assets variable?